phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,216 @@
+package org.apache.phoenix.schema.json;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import org.apache.hadoop.hbase.util.Bytes;
+import org.codehaus.jackson.JsonFactory;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParseException;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.JsonParser.Feature;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ValueNode;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * The {@link PhoenixJson} wraps json and uses Jackson library to parse and traverse the json. It
+ * should be used to represent the JSON data type and also should be used to parse Json data and
+ * read the value from it. It always conside the last value if same key exist more than once.
+ */
+public class PhoenixJson {
+    private final JsonNode node;
+    private String jsonAsString;
+
+    /**
+     * Static Factory method to get an {@link PhoenixJson} object. It also validates the json and
+     * throws {@link JsonParseException} if it is invalid with line number and character, which
+     * should be wrapped in {@link SQLException} by caller.
+     * @param data Buffer that contains data to parse
+     * @param offset Offset of the first data byte within buffer
+     * @param length Length of contents to parse within buffer
+     * @return {@link PhoenixJson}.
+     * @throws JsonParseException
+     * @throws IOException
+     */
+    public static PhoenixJson getPhoenixJson(byte[] jsonData, int offset, int length)
+            throws JsonParseException, IOException {
```

Comment:
Probably best to throw a SQLException here with the JsonParseException or IOException nested inside. Perhaps create an INVALID_JSON_DATA exception in SQLExceptionCode that you can use for the JsonParseException.